HW02/Aliaksandr_Korzun/aliaksandr_korzun_hw_02_t_03.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'time'
+
+LOG_FILE = 'logfile_task_3.txt'
+
+SUBSTRING = 'calling core'
+TIME_TEMPLATE = /^\d{4}\-\d{2}\-\d{2}\s\d{2}\:\d{2}\:\d{2}\.\d{1}/.freeze
+
+def duration_of_actions(logs)
+  result_hash = {}
+  # Create array of logs
+  file_data = File.readlines(logs).map(&:chomp)
+                  .delete_if { |line| core?(line) }
+  return 0 if file_data.size < 2
+
+  # Fill in hash
+  file_data.each_cons(2) do |item|
+    result_hash["#{get_action(item.first)}->#{get_action(item.last)}"] =
+      (parse_time(item.last) - parse_time(item.first)).to_s
+  end
```

Comment:
Try to use `.map` to avoid using `result_hash`
```suggestion
  file_data.map.each_cons(2) do |item|
    result_hash["#{get_action(item.first)}->#{get_action(item.last)}"] =
      (parse_time(item.last) - parse_time(item.first)).to_s
  end
```

Reply:
But if I use -  file_data.map.each_cons(2) do |item|
method returns nil
At the same time if I use file_data.each_cons(2).map do |item|
method returns just array ["49.1", "40"] instead hash

Reply:
Why do you need hash at all? As far as I know, you just need to return time differences in array

Reply:
Got it. Just wanted to return time differences with actions.